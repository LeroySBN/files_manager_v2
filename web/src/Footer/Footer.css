:root {
  --primary-color: #e1003c;
  --border-style: 3px solid var(--primary-color);
  --primary-font: Courier New, Courier, monospace;
  --secondary-font: Helvetica, sans-serif;
  --tertiary-font: Lora, serif;
}

.App-footer {
  position: static;
  bottom: 0;
  border-top: var(--border-style);
  padding: 1.2rem;
  font-size: 1rem;
  text-align: center;
  font-style: italic;
  background-color: white;
  z-index: 1;
  width: calc(100% - 2em);
}
