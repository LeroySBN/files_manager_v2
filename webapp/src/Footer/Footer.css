:root {
  --primary-color: #e1003c;
  --border-style: 3px solid var(--primary-color);
  --primary-font: Courier New, Courier, monospace;
  --secondary-font: Helvetica, sans-serif;
  --tertiary-font: Lora, serif;
}

.App-footer {
  position: sticky;
  bottom: 0;
  border-top: var(--border-style);
  font-style: italic;
  background-color: white;
  z-index: 1;
  width: 100%;
}

.footerContainer {
  font-size: 0.8rem;
  text-align: center;
  padding: 1rem;
}
