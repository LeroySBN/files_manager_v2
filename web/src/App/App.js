import React from 'react';
import { connect } from 'react-redux';
import $ from 'jquery';
import { StyleSheet, css } from 'aphrodite';
import Notifications from '../Notifications/Notifications';
import Header from '../Header/Header';
import { Login } from '../Login/Login';
import Footer from '../Footer/Footer';
import CourseList from '../CourseList/CourseList';
import { getLatestNotification } from '../utils/utils';
import BodySectionWithMarginBottom from '../BodySection/BodySectionWithMarginBottom';
import BodySection from '../BodySection/BodySection';
import { user, AppContext } from './AppContext';
import { uiReducer } from '../reducers/uiReducer';

const listCourses = [
  { id: 1, name: "ES6", credit: 60 },
  { id: 2, name: "Webpack", credit: 20 },
  { id: 3, name: "React", credit: 40 },
];

const listNotifications = [
  { id: 1, type: "default", value: "New course available" },
  { id: 2, type: "urgent", value: "New resume available" },
  { id: 3, type: "urgent", html: getLatestNotification() },
];

export class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displayDrawer: false,
      user: user,
      logOut: this.logOut,
      listNotifications: listNotifications
    };

    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleDisplayDrawer = this.handleDisplayDrawer.bind(this);
    this.handleHideDrawer = this.handleHideDrawer.bind(this);
    this.logIn = this.logIn.bind(this);
    this.logOut = this.logOut.bind(this);
    this.markNotificationAsRead = this.markNotificationAsRead.bind(this);
    
  }

  componentDidMount() {
    $(document).on('keydown', this.handleKeyPress);
  }

  componentWillUnmount() {
    $(document).off('keydown', this.handleKeyPress);
  }
  
  handleKeyPress = (event)=> {
    if (event.ctrlKey && event.key === 'h') {
      alert('Logging you out');
      this.state.logOut();
    }
  }

  handleDisplayDrawer = ()=> {
    // console.log('handleDisplayDrawer called');
    this.setState({ displayDrawer: true });
  }

  handleHideDrawer = ()=> {
    this.setState({ displayDrawer: false });
  }

  logIn = (email, password)=> {
    this.setState({ user: { email, password, isLoggedIn: true }})
  }

  logOut = ()=> {
    this.setState({ user: user})
  }

  markNotificationAsRead = (id)=> {
    this.setState({
      listNotifications: this.state.listNotifications.filter((notification)=> notification.id !== id)
    });
  };

  render() {
    return (
      <AppContext.Provider value={{
        user: this.state.user,
        logOut: this.state.logOut,
      }}>
        <React.Fragment>
          <Notifications 
            displayDrawer={this.state.displayDrawer}
            listNotifications={this.state.listNotifications}
            handleDisplayDrawer={this.handleDisplayDrawer}
            handleHideDrawer={this.handleHideDrawer}
            markNotificationAsRead={this.markNotificationAsRead}
          />
          <div className={css(styles.App)}>
            <Header />
            {this.state.user.isLoggedIn ? ( 
              <BodySectionWithMarginBottom title="Course list">
                <CourseList listCourses={listCourses} />
              </BodySectionWithMarginBottom>
            ) : ( 
              <BodySectionWithMarginBottom title="Log in to continue">
                <Login logIn={this.logIn} />
              </BodySectionWithMarginBottom>
            ) }
            <BodySection title="News from the school">
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
                Donec mollis non risus quis semper. Donec ac lectus in magna 
                faucibus lobortis. Morbi sed accumsan arcu. Pellentesque 
                habitant morbi tristique senectus et netus et malesuada fames 
                ac turpis egestas. Donec pellentesque tincidunt facilisis. 
                Nam eu mauris enim. Nullam et bibendum ipsum, eget lobortis 
                metus. In congue, nunc sed pretium porta, ligula lacus 
                sollicitudin lorem, in pellentesque metus est quis arcu. 
                Nulla imperdiet eget dui nec consequat. Etiam non consequat ex. 
                Nullam eget accumsan ante, eu lacinia sapien. Integer vel 
                tincidunt nunc. Class aptent taciti sociosqu ad litora torquent 
                per conubia nostra, per inceptos himenaeos.
              </p>
            </BodySection>
            <Footer />
          </div>
        </React.Fragment>
        </AppContext.Provider>
    );
  }
}

const styles = StyleSheet.create({
  App: {
    minHeight: '100vh',
    fontFamily: 'Courier New, Courier, monospace',
    margin: '0 8px',
  },
});

export function mapStateToProps (state) {
  return {
    isLoggedIn: state.ui.isUserLoggedIn,
  };
}

export default connect(mapStateToProps, null)(App);
