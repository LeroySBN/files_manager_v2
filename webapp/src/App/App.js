import React from 'react';
import PropTypes from 'prop-types';
import {css, StyleSheet} from 'aphrodite';
import {connect} from 'react-redux';
import {checkAuth, clearAuthMessage} from '../actions/authActions';
import {toggleNotificationDrawer} from "../actions/uiActions";
import {Login} from '../Login/Login';
import {Signup} from '../Signup/Signup';
import Logo from '../Logo/Logo';
import {Header} from "../Header/Header";
import {Footer} from "../Footer/Footer";
// import Notifications from "../Notifications/Notifications";
import BodySectionWithMarginBottom from "../BodySection/BodySectionWithMarginBottom";
import {FileList} from "../FileList/FileList";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoginView: true,
    };
  }

  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyPress);
    this.props.checkAuth();
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyPress);
  }

  handleKeyPress = (event) => {
    if (event.ctrlKey && event.key === 'h') {
      alert('Logging you out');
      this.props.logout();
    }
  };

  toggleToSignup = () => {
    this.props.clearAuthMessage();
    this.setState({ isLoginView: false });
  };

  toggleToLogin = () => {
    this.props.clearAuthMessage();
    this.setState({ isLoginView: true });
  };

  render() {
    const { isLoggedIn, dashboardTitle } = this.props;
    const { isLoginView } = this.state;

    return (
      <div className={css(styles.App)}>
        {isLoggedIn ? (
            <React.Fragment>
              {/*<Notifications*/}
              {/*    displayDrawer={this.state.displayDrawer}*/}
              {/*    handleDisplayDrawer={this.handleDisplayDrawer}*/}
              {/*    handleHideDrawer={this.handleHideDrawer}*/}
              {/*    markNotificationAsRead={this.markNotificationAsRead}*/}
              {/*/>*/}
              <div className={css(styles.App)}>
                <Header />
                <BodySectionWithMarginBottom title={dashboardTitle}>
                  <FileList />
                </BodySectionWithMarginBottom>
                <Footer />
              </div>
            </React.Fragment>
        ) : (
          <div className={css(styles.AccessContainer)}>
            <Logo />
            {isLoginView ? (
              <Login showSignup={this.toggleToSignup} />
            ) : (
              <Signup showLogin={this.toggleToLogin} />
            )}
          </div>
        )}
      </div>
    );
  }
}

const styles = StyleSheet.create({
  App: {
    fontFamily: 'Inter, sans-serif',
    background: 'linear-gradient(0deg, #3d85c677, #fff8f8)',
    minHeight: '100vh',
    margin: 0,
    padding: 0,
  },
  Dashboard: {
    position: 'relative',
    maxWidth: '1200px',
    margin: '0 auto',
    padding: '20px',
  },
  AccessContainer: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    gap: '2rem',
    padding: '2rem',
    borderRadius: '8px',
    backgroundColor: 'transparent',
  },
});

App.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  checkAuth: PropTypes.func,
  clearAuthMessage: PropTypes.func,
  toggleNotificationDrawer: PropTypes.func,
  dashboardTitle: PropTypes.string.isRequired
};

App.defaultProps = {
  isLoggedIn: false,
};

const mapStateToProps = (state) => ({
  isLoggedIn: state.auth.get('isLoggedIn'),
  displayDrawer: state.ui.get('isNotificationDrawerVisible'),
  dashboardTitle: state.ui.get('dashboardFocus')
});

const mapDispatchToProps = {
  checkAuth,
  clearAuthMessage,
  toggleNotificationDrawer
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
