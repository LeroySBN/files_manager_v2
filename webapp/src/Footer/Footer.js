import React from 'react';
import './Footer.css';
import {getFullYear} from '../utils/utils';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import WithLogging from "../HOC/WithLogging";
import {css} from "aphrodite";

function Footer({appName}) {
  return (
    <>
      <div className="App-footer">
        <div className="footerContainer container">
          <p>Copyright {getFullYear()} - {appName}</p>
        </div>
        </div>
    </>
  );
}

Footer.propTypes = {
  appName: PropTypes.string.isRequired,
}

const mapStateToProps = (state) => ({
  appName: state.ui.get('appName'),
});

const ConnectedFooter = connect(mapStateToProps, null)(Footer);
const LoggedFooter = WithLogging(ConnectedFooter);

export { LoggedFooter as Footer };
