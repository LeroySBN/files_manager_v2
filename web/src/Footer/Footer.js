import React from 'react';
import './Footer.css';
import { getFooterCopy, getFullYear } from '../utils/utils';
import { AppContext } from '../App/AppContext';

export default function Footer() {
  const { user } = React.useContext(AppContext);
  return (
    <>
      <div className="App-footer">
        {user.isLoggedIn && <a href='#'>Contact us</a>}
        <p>Copyright {getFullYear()} - {getFooterCopy()}</p>
        </div>
    </>
  );
}
