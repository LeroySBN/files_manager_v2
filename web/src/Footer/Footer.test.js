import React from 'react';
import { shallow, mount } from 'enzyme';
import Footer from './Footer';
import { StyleSheetTestUtils } from "aphrodite";
import { AppContext } from '../App/AppContext';

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection();
});

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
});


describe('Footer component testing', () => {
    it('Footer renders without crashing', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Footer renders a div with the class App-footer', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.find('.App-footer').text()).toContain('Copyright');
  });

  it("Tests that there is no link rendered when user is logged out within context", () => {
    const context = {
      user: {
        email: "",
        password: "",
        isLoggedIn: false,
      },
    };

    const wrapper = mount(
      <AppContext.Provider value={context}>
        <Footer />
      </AppContext.Provider>
    );

    expect(wrapper.find("a").length).toBe(0);
    expect(wrapper.find("a").exists()).toBe(false);
    expect(wrapper.text()).not.toContain("Contact us");

    wrapper.unmount();
  });

  it("Tests that there is a link rendered when user is logged in within context", () => {
    const context = {
      user: {
        email: "",
        password: "",
        isLoggedIn: true,
      },
    };

    const wrapper = mount(
      <AppContext.Provider value={context}>
        <Footer />
      </AppContext.Provider>
    );

    expect(wrapper.find("a").length).toBe(1);
    expect(wrapper.find("a").exists()).toBe(true);
    expect(wrapper.text()).toContain("Contact us");

    wrapper.unmount();
  });
});
