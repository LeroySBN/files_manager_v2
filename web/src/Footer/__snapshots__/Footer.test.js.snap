// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component testing Footer renders without crashing 1`] = `ShallowWrapper {}`;
